ptx/src/test/spirv_run/mod.rs

```diff
@@ -224,6 +238,34 @@ fn test_hip_assert<
     Ok(())
 }
 
+fn test_llvm_assert<
+    'a,
+>(
+    name: &str,
+    ptx_text: &'a str,
+    expected_ll: &str
+) -> Result<(), Box<dyn error::Error + 'a>> {
+    let ast = ptx_parser::parse_module_checked(ptx_text).unwrap();
+    let llvm_ir = pass::to_llvm_module(ast).unwrap();
+    let actual_ll = llvm_ir.llvm_ir.print_as_asm();
+    let result = catch_unwind(||
+        pretty_assertions::assert_eq!(actual_ll, expected_ll));
+    if let Err(cause) = result {
+        // Write actual generated LLVM IR to directory specified by environment variable
+        // TEST_PTX_LLVM_FAIL_DIR if test fails
+        let output_dir = env::var("TEST_PTX_LLVM_FAIL_DIR");
+        if let Ok(output_dir) = output_dir {
+            let output_dir = Path::new(&output_dir);
+            create_dir_all(&output_dir).unwrap();

```

Comment:
(Optional) My preference is that free functions come with the namespace, so `fs::create_dir_all` instead of `create_dir_all`